packages/preferences/src/browser/folder-launch-preference-provider.ts

```diff
@@ -0,0 +1,40 @@
+/********************************************************************************
+ * Copyright (C) 2019 Red Hat, Inc. and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+import { injectable } from 'inversify';
+import URI from '@theia/core/lib/common/uri';
+import { FolderPreferenceProvider } from './folder-preference-provider';
+
+@injectable()
+export class FolderLaunchPreferenceProvider extends FolderPreferenceProvider {
+
+    async getUri(): Promise<URI | undefined> {
+        this.folderUri = new URI(this.options.folder.uri);
+        if (await this.fileSystem.exists(this.folderUri.toString())) {
+            const uri = this.folderUri.resolve('.theia').resolve('launch.json');
```

Comment:
It should respect launch.json from vscode as well. See logic in `DebugConfigurationWidget`, if we move it here it should be properly reimplemented.

Reply:
https://github.com/theia-ide/theia/blob/master/packages/debug/src/browser/debug-configuration-manager.ts#L74

Reply:
I think this should be separate issue and another pull request because there are some things to discuss. For example, if a project has one launch configurations for `Theia` and one for `VScode` which names are same but they differ in other fields. Which one should be returned if a user gets configuration by name? How a user should update or delete a concrete launch configuration?

@akosyakov do you mind if I open an issue?

Reply:
It should work as on master. If a folder has .theia then theia settings is used, if not then VS Code settings.